foundations/messages/state-init.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: "`StateInit` structure"
+---
+
+`StateInit` is a data structure sent as part of a message, either internal or incoming external, used to
+
+- deploy a smart contract, for which it stores the initial code and data of a contract;
+- [unfreeze](/foundations/status#status-transitions) a contract, where its _last_ code and data.
+
+This structure can be appended in-place or as a referenced cell to every message:
+
+```tlb
+message$_
+    {X:Type}
+    info:CommonMsgInfo
+    init:(Maybe (Either StateInit ^StateInit))
+    body:(Either X ^X)
+= Message X;
+```
+
+The `StateInit` structure is defined as follows:
+
+```tlb
+_
+    fixed_prefix_length:(Maybe (## 5))
+    special:(Maybe TickTock)
+    code:(Maybe ^Cell)
+    data:(Maybe ^Cell)
+    library:(Maybe ^Cell)
+= StateInit;
+```
+
+where the fields are:
+
+- `special`: indicating was this message created by the [tick-tock transaction](/foundations/messages/tick-tock) or not.
```

Comment:
afaik, this flag was mean to register a smart contract for tick-tock events, however, it is ignored now